pkg/storage/s3.go

```diff
@@ -307,13 +363,17 @@ func (rs *S3Storage) FileExists(ctx context.Context, file string) (bool, error)
 // The first argument is the file path that can be used in `Open`
 // function; the second argument is the size in byte of the file determined
 // by path.
-func (rs *S3Storage) WalkDir(ctx context.Context, fn func(string, int64) error) error {
+func (rs *S3Storage) WalkDir(ctx context.Context, dir string, listCount int64, fn func(string, int64) error) error {
```

Comment:
As the `listCount` parameter (or maybe also the `dir` param) is only used for s3 implement as a contextal value, I think It's better to put this kind of control param into a abstract `ctx interface{}` field. Maybe there are other similar contextal paramers need for gcs for example?

Reply:
I'll move these parameters in ctx after this PR merged, because this PR is a cherry-pick PR.